internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -600,6 +600,13 @@ func shouldUpdateIntegrationTestGitResolver(integrationTestScenario *v1beta2.Int
 	annotations := snapshot.GetAnnotations()
 	params := resolverParamsToMap(testResolverRef.Params)
 
+	// if target revision sha differs from source revision sha we do not want to overwrite it
+	sourceRevisionSha := annotations[gitops.PipelineAsCodeSHAAnnotation]
```

Comment:
I think we should use the target branch (`pipelinesascode.tekton.dev/branch`) instead of the SHA here.

Reply:
I have finding myself mixed up re: the target/source @dirgim, are you referring to [pipelinesascode.tekton.dev/source-branch](https://github.com/konflux-ci/integration-service/blob/a0e1e6c5feef521f210c5ce613e2b03c38f3132e/tekton/build_pipeline.go#L39C2-L40C83)? How does this value get loaded?

Reply:
@dirgim I guess you mean `pac.test.appstudio.openshift.io/branch` in snapshot_adapter.go because the prefix in `pipelinesascode.tekton.dev/branch` is updated when copying [label/annotation](https://github.com/konflux-ci/integration-service/blob/main/gitops/snapshot.go#L866-L867) , right?
@Josh-Everett I think `/branch` mean the branch that PR will be merged to, and `/source-branch` means the PR is opened from, or you may find some pipelineRun/snapshot and compare them.

Reply:
@hongweiliu17 you are correct, I forgot the prefix change when it's applied to Snapshots. 
You are also right about the `/branch` meaning target branch, which is what we should be considering when deciding if we want to overwrite the IntegrationTestScenarios git reference details.

Reply:
Ok I think I was able to untangle my confusion with your help. I updated the existing unit test to make it reflect the new behavior but I am wondering if it is enough to illustrate the desired effect.